pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix

```diff
@@ -11,48 +14,81 @@
+    nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [makeWrapper];
+
+    # xcpp needs a collection of flags to start up properly, so wrap it by default.
+    # We'll provide the unwrapped version as a passthru
+    flags = cling.flags ++ [
+      "-resource-dir" "${cling.unwrapped}"
+      "-L" "${cling.unwrapped}/lib"
+      "-l" "${cling.unwrapped}/lib/cling.so"
+    ];
+
+    fixupPhase = ''
+      runHook preFixup
+
+      wrapProgram $out/bin/xcpp --add-flags "$flags"
+
+      runHook postFixup
+    '';
+
+    doInstallCheck = true;
+    installCheckPhase = ''
+      runHook preCheck
+
+      # Smoke check: run a test notebook using Papermill by creating a simple kernelspec
+      mkdir -p kernels/cpp17
+      export JUPYTER_PATH="$(pwd)"
+      cat << EOF > kernels/cpp17/kernel.json
       {
-        displayName = "C++" + versionSuffix;
-        argv = [
-          "${xeus-cling}/bin/xcpp"
-        ]
-        ++ cling.flags
-        ++ [
-          "-resource-dir" "${cling.unwrapped}"
-          "-L" "${cling.unwrapped}/lib"
-          "-l" "${cling.unwrapped}/lib/cling.so"
-          "-std=${std}"
-          # "-v"
-          "-f" "{connection_file}"
-        ];
-        language = "cpp";
-        logo32 = fetchurl {
-          url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/ISO_C%2B%2B_Logo.svg/32px-ISO_C%2B%2B_Logo.svg.png";
-          hash = "sha256-cr0TB8/j2mkcFhfCkz9F7ZANOuTlWA2OcWtDcXyOjHw=";
-        };
-        logo64 = fetchurl {
-          url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/ISO_C%2B%2B_Logo.svg/64px-ISO_C%2B%2B_Logo.svg.png";
-          hash = "sha256-nZtJ4bR7GmQttvqEJC9KejOxphrjjxT36L9yOIITFLk=";
-        };
-      };
+        "argv": ["$out/bin/xcpp", "-std=c++17", "-f", "{connection_file}"],
+        "language": "cpp17"
+      }
+      EOF
+
+      ${python3.pkgs.papermill}/bin/papermill ${./test.ipynb} out.ipynb
+      result="$(cat out.ipynb | ${jq}/bin/jq -r '.cells[0].outputs[0].text[0]')"
+      if [[ "$result" != "Hello world." ]]; then
+        echo "Kernel test gave '$result'. Expected: 'Hello world.'"
+        exit 1
+      fi
+
+      runHook postCheck
+    '';
+
+    passthru = (oldAttrs.passthru or {}) // {
+      unwrapped = xeus-cling-unwrapped;
+    };
+  });
+
+  mkKernelSpec = std: {
+    displayName = builtins.replaceStrings ["c++"] ["C++ "] std;
+    argv = [
+      "${xeus-cling}/bin/xcpp"
+      "-std=${std}"
+      "-f" "{connection_file}"
+    ];
+    language = "cpp";
+    logo32 = fetchurl {
+      url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/ISO_C%2B%2B_Logo.svg/32px-ISO_C%2B%2B_Logo.svg.png";
+      hash = "sha256-cr0TB8/j2mkcFhfCkz9F7ZANOuTlWA2OcWtDcXyOjHw=";
+    };
+    logo64 = fetchurl {
+      url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/ISO_C%2B%2B_Logo.svg/64px-ISO_C%2B%2B_Logo.svg.png";
+      hash = "sha256-nZtJ4bR7GmQttvqEJC9KejOxphrjjxT36L9yOIITFLk=";
```

Comment:
should be "sha256-7SjOcSaSPUHIKnjBxMdn+KSjviL69IXhX7eJsacYeGE="

Reply:
Done